DM-13871: Talk about Slack

srp3rd opened the pull request:
Added text about advantages instant messaging, our transition from HipChat to Slack

srp3rd committed:
Added text about instant messaging tools

srp3rd committed:
Add author and location

timj commented on slack.tex:
```diff
@@ -1,7 +1,53 @@
+continue to use today. The move from HipChat to Slack was straightforward,
+although we did lose the discussion history from HipChat, and Slack initially
+didn't have video chat (which it does now). We also took the opportunity to
+normalize the users' chat names to match the names used on GitHub, which
```
This didn't work. We now ask that if they are different people can associate the github ID with their slack account and Jenkins bot will translate.

srp3rd replied:
Ok, a couple of options here;  I can mention our intent and subsequent solution, or leave this out.  If I keep it in and explain, I have a question:  Why didn't this work?

timj replied:
I think people who came late to Github didn't get the name they wanted and saw no reason why they should be forced to use that name on Slack when they were able to choose anything they wanted.

timj commented on slack.tex:
```diff
@@ -1,7 +1,53 @@
+conversations in a team-wide tool has been very beneficial. Discussions happen 
+with more team members participating, instead of just who happened to be on a 
+phone call or an e-mail thread. Questions and concerns have been able to be 
+answered much more quickly.
```
Maybe say that we have some channels dedicated to non-work topics to allow friendships and team building to happen outside of the work context. We also have public channels for each site to allow local discussions

srp3rd replied:
I was wondering whether to add something about the non-work channels, since I wasn't sure of the "voice" of the paper.  I'm adding this in now.

timj commented on slack.tex:
```diff
@@ -1,7 +1,53 @@
+
+After a brief testing period, we decided to transition to Slack, which we
+continue to use today. The move from HipChat to Slack was straightforward,
+although we did lose the discussion history from HipChat, and Slack initially
```
99% of history should be lost. I can't remember searching history in Slack for anything older than a couple of weeks. I don't think the history from two years ago is as important as we think it is. It might be worth mentioning that we have had difficulties balancing when to "archive" a conversation onto community.

srp3rd replied:
OK, I addressed this in the update.

timj commented on slack.tex:
```diff
@@ -1,7 +1,53 @@
+instead of creating a public chat room for an ephemeral discussion topic. 
+Previews of URLs, global and per channel notification preferences and an 
+API to integrate third-party or custom chat-bot apps. One of those apps, 
+qrbot, is discussed in the next section.  
```
sqrbot.

srp3rd replied:
heh....I deleted that character in the process of reformatting this to < 80 character lines.  I'll fix it.

timj replied:
For latex I prefer one sentence per line (no artificial breaks at 80 chars). That makes it easier for git to know what changed. Otherwise you end up making one change to the start of a paragraph and see the entire paragraph turn up in the diff as it is reformatted.

timj commented on slack.tex:
```diff
@@ -1,7 +1,53 @@
+application was last used, or by marking them already read.
+
+Other features include direct messaging can be set up for small groups,
+instead of creating a public chat room for an ephemeral discussion topic. 
```
Maybe also mention that we create channels specifically to follow topics in conferences and workshops to allow involvement from people who aren't attending.

srp3rd replied:
That was the intention of the last part of the first line of the section, but I'll make it more clear here.

srp3rd replied:
I'm also adding the fact that we're coordinating this paper through a Slack channel.

timj commented on spie-10707-10.tex:
```diff
@@ -28,6 +29,8 @@
 \affil[c]{Centre de Calcul de l'IN2P3, USR 6402 du CNRS/IN2P3, 43 Bd. du 11 Novembre 1918, 69622 Villeurbanne Cedex, France}
 \affil[d]{SLAC National Accelerator Laboratory, 2575 Sand Hill Rd, Menlo Park, CA 94025, USA}
 \affil[e]{Department of Astrophysical Sciences, Princeton University, Princeton, NJ 08544}
+\affil[f]{NCSA, University of Illinois at Urbana-Champaign, 1205 W. Clark St. Urbana, IL 61801}
```
I think you are technically "e" affiliation because you come before Pim.

srp3rd replied:
fixed

timj commented on slack.tex:
```diff
@@ -1,7 +1,53 @@
+has several options on how unread messages are handled and displayed, allowing
+users to choose how the application behaves when the application restarts. 
+Users can decide how to handle unread messages either showing them where the
+application was last used, or by marking them already read.
```
We also use adhoc video calls that can be created for a channel. This is easier than looking up a Blue Jeans number.

srp3rd replied:
Added a section on this.

timj reviewed (changes requested):
Thanks. I've made some comments. It's hard to remember what HipChat was like :-)

srp3rd committed:
Change affliate ordering

srp3rd committed:
Additional text for:
 - explaining the github id/slack account jenkins bot translation
 - non-work topic channels
 - local channels
 - archiving history
 - typo for "sqrbot"
 - conference/workshop channels and this paper's channel

srp3rd commented:
Made all the requested additions/changes.

timj commented on slack.tex:
```diff
@@ -1,7 +1,77 @@
+although we did lose the discussion history from HipChat.  At first, we 
+thought this would be an issue, but in practice, we rarely search history older
+than a couple of weeks.  We have had difficulties in balancing when to 
+archive these discussions for the community.  We've also had some problems
```
By "community" I meant "when to formally migrate a discussion to community.lsst.org where a decision can be more easily disseminated."

srp3rd replied:
Before I commit this, is this more like what you'd like to see?  "We have had difficulties in balancing when to formally migrate discussions made in these channels so we can more easily disseminate decisions to community.lsst.org."

timj replied:
Something like that. We can edit later. The point being that sometimes the five people in the channel come up with some consensus on some interesting topic but you can't leave it sitting there in the channel history. You either need to make a comment on Jira or write a community post summarizing the discussion.

timj commented on slack.tex:
```diff
@@ -1,7 +1,77 @@
+conferences, which helps team members coordinate while on-site and to involve
+people who aren't attending.  This paper itself was even organized
+within a Slack channel, allowing us to discuss it and to get automatic
+notifications of changes to it.
```
"via the GitHub Slack bot."

srp3rd replied:
added

srp3rd committed:
Added text about decision migration and to clarify that GitHub Slack bot
is making notifications
